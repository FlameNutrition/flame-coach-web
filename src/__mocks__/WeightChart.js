// FIXME: Try to understand why __mock__ folder doesn't work
const WeightChart = () => 'test';

export default WeightChart;
