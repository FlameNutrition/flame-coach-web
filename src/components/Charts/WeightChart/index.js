import PropTypes from 'prop-types';
import React, { useState } from 'react';
import clsx from 'clsx';
import '../../../../node_modules/react-vis/dist/style.css';
import {
  FlexibleXYPlot,
  Hint,
  HorizontalGridLines,
  LineMarkSeries,
  VerticalGridLines,
  XAxis,
  YAxis
} from 'react-vis';
import moment from 'moment';
import { maxTicksLimit, maxTicksLimitMobile, minMaxValue } from '../utils/chartUtil';
import { getTimeRangeDates } from '../../../utils/timeRangeUtil';
import Card from '@material-ui/core/Card';
import makeStyles from '@material-ui/styles/makeStyles';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';

const useStyles = makeStyles(() => ({
  root: {},
  chart: {
    display: 'flex',
    height: '100%',
    width: '100%'
  },
  weightGraphicTitle: {
    textAlign: 'center',
    paddingBottom: '10px'
  }
}));

const WeightChart = ({
  isMobile,
  timeFrame,
  className,
  dataChart,
  measureUnit
}) => {
  const classes = useStyles();

  const maxTick = isMobile ? maxTicksLimitMobile(timeFrame) : maxTicksLimit(timeFrame);

  const filterDates = getTimeRangeDates(timeFrame, moment());

  const minMaxWeightRange = minMaxValue(dataChart);

  const dataFormatted = dataChart.map((data) => ({
    y: data.value,
    x: moment(data.date, 'YYYY-MM-DD')
  }));

  const fromDttm = moment(filterDates.fromDttm, 'YYYY-MM-DD');
  const toDttm = moment(filterDates.toDttm, 'YYYY-MM-DD');

  const minWeight = minMaxWeightRange.minValue - 10;
  const maxWeight = minMaxWeightRange.maxValue + 10;

  const [value, setValue] = useState();

  return (
    <Card className={clsx(classes.root)}>
      <CardContent className={className}>
        <Typography className={classes.weightGraphicTitle} component="h2" variant="h5" gutterBottom>
          Weight
        </Typography>
        <Box className={classes.chart}>
          <FlexibleXYPlot
            margin={{
              left: 30,
              right: 30,
              top: 5,
              bottom: isMobile ? 30 : 60
            }}
            xType="time"
            yDomain={[minWeight, maxWeight]}
            xDomain={[fromDttm, toDttm]}
          >
            <VerticalGridLines/>
            <HorizontalGridLines/>
            <XAxis
              tickTotal={maxTick}
              tickSize={0}
              tickPadding={isMobile ? 12 : 10}
              tickFormat={(xValue) => {
                return moment(xValue)
                  .format('MM-DD');
              }}
              tickLabelAngle={isMobile ? -45 : 0}
              style={{
                ticks: { paddingLeft: '50px' },
                text: { fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif' }
              }}
            />
            <YAxis
              tickSizeOuter={0}
              tickSizeInner={0}
              tickPadding={7}
              style={{
                text: { fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif' }
              }}
            />
            <LineMarkSeries
              data={dataFormatted}
              size={3}
              color="#3f51b5"
              fill="#3f51b5"
              onValueMouseOver={(datapoint) => {
                setValue(datapoint);
              }}
              onValueMouseOut={() => setValue(null)}
            />
            {value ? (
              <Hint
                value={value}
                format={(datapoint) => {
                  return [
                    {
                      title: 'Date',
                      value: moment(datapoint.x)
                        .format('YYYY-MM-DD')
                    },
                    {
                      title: 'Weight',
                      value: `${datapoint.y} ${measureUnit}`
                    }];
                }}
                style={{ text: { fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif' } }}
              />
            ) : null}
          </FlexibleXYPlot>
        </Box>
      </CardContent>
    </Card>
  );
};

WeightChart.propTypes = {
  className: PropTypes.string,
  timeFrame: PropTypes.string.isRequired,
  dataChart: PropTypes.array.isRequired,
  isMobile: PropTypes.bool,
  measureUnit: PropTypes.string.isRequired
};

WeightChart.defaultProps = {};

export default WeightChart;
