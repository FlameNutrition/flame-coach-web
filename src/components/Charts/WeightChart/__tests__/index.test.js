import React from 'react';
import WeightChart from '../index';
import { render } from '../../../../testing/test-utils';

describe('<WeightChart />', () => {
  it.skip('create component WeightChart', () => {
    const data = [
      { identifier: 1, date: '2021-04-10', value: 70.5 }
    ];

    const { container } = render(<WeightChart
      dataChart={data}
      timeFrame="1_WEEK"
      measureUnit="Kg"
    />);

    // FIXME: Refactor this to instead of using snapshot, search for specific dates
    expect(container).toMatchSnapshot();
  });
});
