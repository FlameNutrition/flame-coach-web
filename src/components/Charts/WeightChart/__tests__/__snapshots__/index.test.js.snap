// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeightChart /> create component WeightChart 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root makeStyles-weightGraphicTitle-3 MuiTypography-h5 MuiTypography-gutterBottom"
      >
        Weight
      </h2>
      <div
        class="MuiBox-root MuiBox-root-4 makeStyles-chart-2"
      >
        <div
          style="width: 100%; height: 100%;"
        >
          <div
            class="rv-xy-plot "
            style="width: 0px; height: 0px;"
          >
            <svg
              class="rv-xy-plot__inner"
              height="0"
              width="0"
            >
              <g
                class="rv-xy-plot__grid-lines undefined"
                transform="translate(30,5)"
              >
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="-65"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="-8.571428571428571"
                  x2="-8.571428571428571"
                  y1="0"
                  y2="-65"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="-17.142857142857142"
                  x2="-17.142857142857142"
                  y1="0"
                  y2="-65"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="-25.71428571428571"
                  x2="-25.71428571428571"
                  y1="0"
                  y2="-65"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="-34.285714285714285"
                  x2="-34.285714285714285"
                  y1="0"
                  y2="-65"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="-42.85714285714286"
                  x2="-42.85714285714286"
                  y1="0"
                  y2="-65"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="-51.42857142857142"
                  x2="-51.42857142857142"
                  y1="0"
                  y2="-65"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="-60"
                  x2="-60"
                  y1="0"
                  y2="-65"
                />
              </g>
              <g
                class="rv-xy-plot__grid-lines undefined"
                transform="translate(30,5)"
              >
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="0"
                  x2="-60"
                  y1="-65"
                  y2="-65"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="0"
                  x2="-60"
                  y1="-48.75"
                  y2="-48.75"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="0"
                  x2="-60"
                  y1="-32.5"
                  y2="-32.5"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="0"
                  x2="-60"
                  y1="-16.25"
                  y2="-16.25"
                />
                <line
                  class="rv-xy-plot__grid-lines__line"
                  x1="0"
                  x2="-60"
                  y1="0"
                  y2="0"
                />
              </g>
              <g
                class="rv-xy-plot__axis rv-xy-plot__axis--horizontal "
                transform="translate(30,-60)"
              >
                <line
                  class="rv-xy-plot__axis__line"
                  x1="0"
                  x2="-60"
                  y1="0"
                  y2="0"
                />
                <g
                  class="rv-xy-plot__axis__ticks"
                  transform="translate(0, 0)"
                >
                  <g
                    class="rv-xy-plot__axis__tick"
                    style="padding-left: 50px;"
                    transform="translate(0, 0)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      style="padding-left: 50px;"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.72em"
                      style="padding-left: 50px; font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="middle"
                      transform="translate(0, 10)"
                    >
                      05-22
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    style="padding-left: 50px;"
                    transform="translate(-8.571428571428571, 0)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      style="padding-left: 50px;"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.72em"
                      style="padding-left: 50px; font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="middle"
                      transform="translate(0, 10)"
                    >
                      05-23
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    style="padding-left: 50px;"
                    transform="translate(-17.142857142857142, 0)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      style="padding-left: 50px;"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.72em"
                      style="padding-left: 50px; font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="middle"
                      transform="translate(0, 10)"
                    >
                      05-24
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    style="padding-left: 50px;"
                    transform="translate(-25.71428571428571, 0)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      style="padding-left: 50px;"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.72em"
                      style="padding-left: 50px; font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="middle"
                      transform="translate(0, 10)"
                    >
                      05-25
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    style="padding-left: 50px;"
                    transform="translate(-34.285714285714285, 0)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      style="padding-left: 50px;"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.72em"
                      style="padding-left: 50px; font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="middle"
                      transform="translate(0, 10)"
                    >
                      05-26
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    style="padding-left: 50px;"
                    transform="translate(-42.85714285714286, 0)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      style="padding-left: 50px;"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.72em"
                      style="padding-left: 50px; font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="middle"
                      transform="translate(0, 10)"
                    >
                      05-27
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    style="padding-left: 50px;"
                    transform="translate(-51.42857142857142, 0)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      style="padding-left: 50px;"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.72em"
                      style="padding-left: 50px; font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="middle"
                      transform="translate(0, 10)"
                    >
                      05-28
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    style="padding-left: 50px;"
                    transform="translate(-60, 0)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      style="padding-left: 50px;"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.72em"
                      style="padding-left: 50px; font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="middle"
                      transform="translate(0, 10)"
                    >
                      05-29
                    </text>
                  </g>
                </g>
              </g>
              <g
                class="rv-xy-plot__axis rv-xy-plot__axis--vertical "
                transform="translate(0,5)"
              >
                <line
                  class="rv-xy-plot__axis__line"
                  x1="30"
                  x2="30"
                  y1="0"
                  y2="-65"
                />
                <g
                  class="rv-xy-plot__axis__ticks"
                  transform="translate(30, 0)"
                >
                  <g
                    class="rv-xy-plot__axis__tick"
                    transform="translate(0, -65)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.32em"
                      style="font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="end"
                      transform="translate(-7, 0)"
                    >
                      60
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    transform="translate(0, -48.75)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.32em"
                      style="font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="end"
                      transform="translate(-7, 0)"
                    >
                      65
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    transform="translate(0, -32.5)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.32em"
                      style="font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="end"
                      transform="translate(-7, 0)"
                    >
                      70
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    transform="translate(0, -16.25)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.32em"
                      style="font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="end"
                      transform="translate(-7, 0)"
                    >
                      75
                    </text>
                  </g>
                  <g
                    class="rv-xy-plot__axis__tick"
                    transform="translate(0, 0)"
                  >
                    <line
                      class="rv-xy-plot__axis__tick__line"
                      x1="0"
                      x2="0"
                      y1="0"
                      y2="0"
                    />
                    <text
                      class="rv-xy-plot__axis__tick__text"
                      dy="0.32em"
                      style="font-family: \\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif;"
                      text-anchor="end"
                      transform="translate(-7, 0)"
                    >
                      80
                    </text>
                  </g>
                </g>
              </g>
              <g
                class="rv-xy-plot__series rv-xy-plot__series--linemark"
              >
                <path
                  class="rv-xy-plot__series rv-xy-plot__series--line "
                  d="M360,-30.875Z"
                  style="opacity: 1; stroke: #3f51b5;"
                  transform="translate(30,5)"
                />
                <g
                  class="rv-xy-plot__series rv-xy-plot__series--mark "
                  transform="translate(30,5)"
                >
                  <circle
                    cx="360"
                    cy="-30.875"
                    r="3"
                    style="opacity: 1; stroke: #3f51b5; fill: #3f51b5; stroke-width: 1;"
                  />
                </g>
              </g>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
